moveit_planners/chomp/chomp_motion_planner/src/chomp_trajectory.cpp

```diff
@@ -253,4 +253,63 @@ void ChompTrajectory::fillInMinJerk()
   }
 }
 
+void ChompTrajectory::fillInFromTrajectory(moveit_msgs::MotionPlanDetailedResponse& res)
+{
+  // create a RobotTrajectory msg to obtain the trajectory from the MotionPlanDetailedResponse object
+  moveit_msgs::RobotTrajectory trajectory_msgs = res.trajectory[0];  // = new moveit_msgs::MotionPlanResponse();
+
+  // get the default number of points in the CHOMP trajectory
+  int num_chomp_trajectory_points = (*this).getNumPoints();
+  // get the number of points in the response trajectory obtained from OMPL
+  int num_response_points = trajectory_msgs.joint_trajectory.points.size();
+  // get the number of joints for each robot state
+  int num_joints_trajectory = trajectory_msgs.joint_trajectory.points[0].positions.size();
+
+  // variables for populating the CHOMP trajectory with correct number of trajectory points
+  int repeated_factor = num_chomp_trajectory_points / num_response_points;
+  int repeated_balance_factor = num_chomp_trajectory_points % num_response_points;
+
+  int trajectory_row_index = 0;
+  if (num_chomp_trajectory_points >= num_response_points)
+  {
+    for (int i = 0; i < num_response_points; i++)
+    {
+      // following for loop repeats each OMPL trajectory pose/row 'repeated_factor' times; alternatively, there could
+      // also
+      // be a linear interpolation between these points later if required
+      for (int k = 0; k < repeated_factor; k++)
```

Comment:
move this inner for loop into a helper function, otherwise its way too nested.... if, for, for, for

Reply:
changed to addition of a new helped method [here](https://github.com/ros-planning/moveit/pull/1012/commits/098c130924ce8c4c7979f48ca04f98ad7d42d244#diff-efa4d0cbd023f4d765a82a694c2571feR314) now its only if, for, for ..{function_call}..